corehq/apps/userreports/tests/test_pillow.py

```diff
@@ -254,6 +258,75 @@ def test_process_doc_from_sql_stale(self):
 
 
 @override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
+class ReuseEvaluationContextTest(TestCase):
+    domain = 'bug-domain'
+
+    @softer_assert()
+    def setUp(self):
+        config1 = get_data_source_with_related_doc_type()
+        config1.save()
+        config2 = get_data_source_with_related_doc_type()
+        config2.table_id = 'other-config'
+        config2.save()
+        self.configs = [config1, config2]
+        self.adapters = [get_indicator_adapter(c) for c in self.configs]
+
+        # one pillow that has one config, the other has both configs
+        self.pillow1 = get_kafka_ucr_pillow(topics=['case-sql'])
+        self.pillow2 = get_kafka_ucr_pillow(topics=['case-sql'])
+        self.pillow1.bootstrap(configs=[config1])
+        self.pillow2.bootstrap(configs=self.configs)
+        with trap_extra_setup(KafkaUnavailableError):
+            self.pillow1.get_change_feed().get_latest_offsets()
+            self.pillow2.get_change_feed().get_latest_offsets()
+
+    def tearDown(self):
+        for adapter in self.adapters:
+            adapter.drop_table()
+            adapter.config.delete()
+        delete_all_cases()
+        delete_all_xforms()
+
+    def _post_case_blocks(self, iteration=0):
+        return post_case_blocks(
+            [
+                CaseBlock(
+                    create=iteration == 0,
+                    case_id='parent-id',
+                    case_name='parent-name',
+                    case_type='bug',
+                    update={'update-prop-parent': iteration},
+                ).as_xml(),
+                CaseBlock(
+                    create=iteration == 0,
+                    case_id='child-id',
+                    case_name='child-name',
+                    case_type='bug-child',
+                    index={'parent': ('bug', 'parent-id')},
+                    update={'update-prop-child': iteration}
+                ).as_xml()
+            ], domain=self.domain
+        )
+
+    def _test_pillow(self, pillow, since):
+        with self.assertNumQueries(15):
```

Comment:
Reduced this to 4 per case here: https://github.com/dimagi/commcare-hq/pull/15611